homeassistant/components/wolflink/config_flow.py

```diff
@@ -0,0 +1,79 @@
+"""Config flow for Wolf SmartSet Service integration."""
+import logging
+
+from httpcore._exceptions import ConnectError
+import voluptuous as vol
+from wolf_smartset.token_auth import InvalidAuth
+from wolf_smartset.wolf_client import WolfClient
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_PASSWORD, CONF_USERNAME
+
+from .const import DOMAIN  # pylint:disable=unused-import
+
+_LOGGER = logging.getLogger(__name__)
+
+USER_SCHEMA = vol.Schema(
+    {vol.Required(CONF_USERNAME): str, vol.Required(CONF_PASSWORD): str}
+)
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Wolf SmartSet Service."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_CLOUD_POLL
+
+    def __init__(self):
+        """Initialize with empty username and password."""
+        self.username = None
+        self.password = None
+        self.hub = None
+        self.fetched_systems = None
+
+    async def async_step_user(self, user_input=None):
+        """Handle the initial step to get connection parameters."""
+        errors = {}
+        if user_input is not None:
+            try:
+                self.hub = WolfClient(
+                    user_input[CONF_USERNAME], user_input[CONF_PASSWORD]
+                )
+                self.fetched_systems = await self.hub.fetch_system_list()
+                self.username = user_input[CONF_USERNAME]
+                self.password = user_input[CONF_PASSWORD]
+                return await self.async_step_device()
+            except ConnectError:
+                errors["base"] = "cannot_connect"
+            except InvalidAuth:
+                errors["base"] = "invalid_auth"
+            except Exception:  # pylint: disable=broad-except
```

Comment:
@MartinHjelmare I have a question. I guess here I should also add some specific exception, right? This is only place where I do not have tests.

Reply:
It's ok to have a broad exception as fallback in that place since we catch the specific expected exceptions first.

Reply:
So the question is: how should I test it. Is patching `self.hub.fetch_system_list()` with `side_effect=Exception` is a good idea?

Reply:
Yes that's good.